### ISSUE TITLE
Monochrome: using shapes/patterns instead of colors?

### ISSUE THREAD

**Issue by HenrikBengtsson** [janpaul123/paperprograms#1]
Instead of color coding programs, do you think it would be possible to encode them by using different orthogonal, distinct shapes and fill patterns as an alternative, e.g. solid squares, empty square, triangles, upside-down triangles, plus symbol, and so on?

That would lower the entrance barrier to those who only got a monochrome printer at home.

PS. Thanks for sharing this and making it open source. I've stumbled upon the Dynamicland website and Twitter feed, but failed to get ahold of DIY info or source code, so I was happy to see yours.

**Comment by forresto** [comment]
https://archive.artoolkit.org/documentation/doku.php?id=3_Marker_Training:marker_barcode has info on "2d barcode" monochrome trackers for AR. The dots are better-suited for framing a page though.

**Comment by blakebjorn** [comment]
Using black circles and filling in a percentage of the interior would probably work much better.

If each circle had 6 'pie pieces' which could be either empty, or black, you could use otsu's binarization to classify each piece as 0, 0.17, 0.33, 0.5, 0.66, 0.83, 1.0

This means with 4 circles, one in each corner, there would be a much greater combination of pieces than the current limit of 600 (although with 5X5X4 corners you should have much more than 600 combinations). Also black and white image processing will be much faster as there is only one array to operate on.

**Comment by forresto** [comment]
One issue that I noticed at DL is that some projected graphics can obscure tracking, especially as the sun sets, making the projections brighter relative to the room. Do you have any idea if monochrome tracking dots would be more or less susceptible to this?

**Comment by janpaul123** [comment]
Yeah, I originally experimented with QR codes but the resolution was not high enough. There're some good ideas here though!
